File: 1538-maximum-points-you-can-obtain-from-cards/maximum-points-you-can-obtain-from-cards.py
class Solution:
    def maxScore(self, cardPoints: List[int], k: int) -> int:
        l_sum=0
        r_sum=0
        max_sum=0
        n=len(cardPoints)

        for i in range (0,k):
            l_sum +=cardPoints[i]
        max_sum=l_sum

        r_ind=n-1
        for i in range(k-1,-1,-1):
            l_sum-=cardPoints[i]
            r_sum+=cardPoints[r_ind]
            r_ind-=1
        
            max_sum=max(max_sum,l_sum+r_sum)
        return max_sum

File: 1538-maximum-points-you-can-obtain-from-cards/maximum-points-you-can-obtain-from-cards.java
class Solution {
    public int maxScore(int[] cardPoints, int k) {

        int leftsum =0;
        int rightsum =0;
        int maxsum=0;
        int n=cardPoints.length;

        for (int i=0; i<k; i++){
            leftsum+=cardPoints[i];
        }
        maxsum=Math.max(leftsum,maxsum);

        int right = n-1;
        for (int r=k-1; r>=0; r--){
            leftsum-=cardPoints[r];
            rightsum+=cardPoints[right];
            right--;
            maxsum=Math.max(leftsum+rightsum,maxsum);
        }
        return maxsum;
        
        
    }
}